source/core/simulation_control.cc

```diff
@@ -48,6 +55,116 @@ SimulationControl::is_output_iteration()
+      set_current_time_step(initial_time_step * startup_timestep_scaling);
+    }
+  else if (iteration_number == 2 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf3 &&
+           bdf_start_method == Parameters::SimulationControl::
+                                 BDFStartupMethods::multiple_step_bdf)
+    {
+      assembly_method = Parameters::SimulationControl::TimeSteppingMethod::bdf2;
+      set_suggested_time_step(initial_time_step * startup_timestep_scaling);
+    }
+  else if (iteration_number == 3 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf3 &&
+           bdf_start_method == Parameters::SimulationControl::
+                                 BDFStartupMethods::multiple_step_bdf)
+    {
+      assembly_method = Parameters::SimulationControl::TimeSteppingMethod::bdf3;
+      set_suggested_time_step(initial_time_step *
+                              (1 - 2 * startup_timestep_scaling));
+    }
+  else if (iteration_number == 4 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf3 &&
+           bdf_start_method == Parameters::SimulationControl::
+                                 BDFStartupMethods::multiple_step_bdf)
+    {
+      assembly_method = Parameters::SimulationControl::TimeSteppingMethod::bdf3;
+      set_suggested_time_step(initial_time_step);
+    }
+  else if (iteration_number <= 1 &&
+           (method ==
+            Parameters::SimulationControl::TimeSteppingMethod::bdf2) &&
+           bdf_start_method ==
+             Parameters::SimulationControl::BDFStartupMethods::sdirk_step)
+    {
+      assembly_method =
+        Parameters::SimulationControl::TimeSteppingMethod::sdirk22;
+      set_suggested_time_step(initial_time_step);
+    }
+  else if (iteration_number <= 2 &&
+           (method ==
+            Parameters::SimulationControl::TimeSteppingMethod::bdf3) &&
+           bdf_start_method ==
+             Parameters::SimulationControl::BDFStartupMethods::sdirk_step)
+    {
+      assembly_method =
+        Parameters::SimulationControl::TimeSteppingMethod::sdirk22;
+      set_suggested_time_step(initial_time_step);
+    }
+  else
+    {
+      assembly_method = method;
+    }
+}
+
+Parameters::SimulationControl::TimeSteppingMethod
+SimulationControl::get_method_to_use()
+{
+  if (iteration_number <= 1 &&
+      method == Parameters::SimulationControl::TimeSteppingMethod::bdf2 &&
+      bdf_start_method ==
+        Parameters::SimulationControl::BDFStartupMethods::sdirk_step)
+    {
+      return Parameters::SimulationControl::TimeSteppingMethod::sdirk22;
+    }
+  else if (iteration_number <= 2 &&
+           method == Parameters::SimulationControl::TimeSteppingMethod::bdf3 &&
+           bdf_start_method ==
+             Parameters::SimulationControl::BDFStartupMethods::sdirk_step)
+    {
+      return Parameters::SimulationControl::TimeSteppingMethod::sdirk33;
+    }
+  else
+    {
+      return method;
+    }
+}
```

Comment:
I don't understand, why did you need an update_method and a get_method that both manage a bit the same thing? is it because of the time-step scaling?

Reply:
the two have been merge together